src/main/java/com/twilio/http/Request.java

```diff
@@ -174,9 +174,18 @@ private String buildURL() {
 
                 host = joinIgnoreNull(".", product, targetEdge, targetRegion, domain);
             }
-
-            return new URI(parsedUrl.getProtocol(), parsedUrl.getUserInfo(), host, parsedUrl.getPort(),
-                    parsedUrl.getPath(), parsedUrl.getQuery(), parsedUrl.getRef()).toURL().toString();
+            String getQuery = "";
+            if( parsedUrl.getQuery() != null ) {
+                getQuery = "?" + parsedUrl.getQuery();
+            }
+            String getFragment = "";
+            if( parsedUrl.getRef() != null ) {
+                getFragment = "#" + parsedUrl.getRef();
+            }
+            String constructedURI = new URI(
+                    parsedUrl.getProtocol(), parsedUrl.getUserInfo(), host, parsedUrl.getPort(),
+                    parsedUrl.getPath(), null,null).toString() + getQuery + getFragment;
+            return new URI(constructedURI).toString();
```

Comment:
`new URI().toString()` from a `new URI().toString()`? This looks even more overly-complicated. Can we just drop the usage of `URI` and construct the URL similar to what we do in other languages?

Reply:
I'll give constructing the URL manually a shot.